### DISCUSSION TITLE
Can I install nix without a connection to the net?

### DISCUSSION THREAD

**Discussion by gowwe** [nix-community/nix-on-droid#107] [Q&A]
Can I install nix without a connection to the net?

**Comment by t184256** [comment]
No. Even if you supply a local bootstrap zipball, it will still need a network connection for the initial generation building.

What's your use case, I wonder?

**Reply by gowwe** [reply]
Please provide a fully installed offline copy

**Reply by t184256** [reply]
Yeah, I heard you. Let's forget for a minute that it's not entirely possible because some important bits of the installation process depend on device-specific things like the user id. Imagine that I somehow did what you're asking for.

How would you use that? Nix is a package manager, and packages come from the network.

What's your specific use case?

**Reply by gowwe** [reply]
Yes, I mean to install the entire application to be ready to install packages

**Reply by t184256** [reply]
Alright, keep your secrets. No, this isn't possible, and without a good use case for it I won't consider implementing it.

**Comment by gowwe** [comment]
Can I download and install nix files manually?

في الجمعة، ٢٦ فبراير ٢٠٢١ ١١:٥١ ص Alexander Sosedkin <
notifications@github.com> كتب:

**Reply by t184256** [reply]
Nix packages aren't "nix files" and one doesn't simply "download them". It's closer to a source-based distro, albeit with a binary cache. I'd recommend you to first try NixOS on your desktop to see what is it all about.

**Comment by gowwe** [comment]
I did not mean the packages, I meant bootstrap.zip and a file size of 300
MB to be inside the application

في الجمعة، ٢٦ فبراير ٢٠٢١ ٥:٢٩ م Alexander Sosedkin <
notifications@github.com> كتب:
